homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,372 @@
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wind_speed_average",
+        translation_key="wind_speed_average",
+        icon="mdi:weather-windy",
+        device_class=SensorDeviceClass.WIND_SPEED,
+        native_unit_of_measurement=UnitOfSpeed.KILOMETERS_PER_HOUR,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=2,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wind_direction",
+        translation_key="wind_direction",
+        icon="mdi:compass-outline",
+        native_unit_of_measurement=DEGREE,
+        state_class=SensorStateClass.MEASUREMENT,
+        event_subscriptions=[EVENT_RAPID_WIND, EVENT_OBSERVATION],
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wind_direction_average",
+        translation_key="wind_direction_average",
+        icon="mdi:compass-outline",
+        native_unit_of_measurement=DEGREE,
+        state_class=SensorStateClass.MEASUREMENT,
+        raw_data_conv_fn=lambda raw_data: raw_data.magnitude,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up WeatherFlow sensors using config entry."""
+
+    @callback
+    def async_add_sensor(device: WeatherFlowDevice) -> None:
+        """Add WeatherFlow sensor."""
+        LOGGER.debug("Adding sensors for %s", device)
+
+        sensors: list[WeatherFlowSensorEntity] = [
+            WeatherFlowSensorEntity(
+                device=device,
+                description=description,
+                is_metric=(hass.config.units == METRIC_SYSTEM),
+            )
+            for description in SENSORS
+            if hasattr(device, description.key)
+        ]
+
+        async_add_entities(sensors)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            format_dispatch_call(config_entry),
+            async_add_sensor,
+        )
+    )
+
+
+class WeatherFlowSensorEntity(SensorEntity):
+    """Defines a WeatherFlow sensor entity."""
+
+    entity_description: WeatherFlowSensorEntityDescription
+    _attr_should_poll = False
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        device: WeatherFlowDevice,
+        description: WeatherFlowSensorEntityDescription,
+        is_metric: bool = True,
+    ) -> None:
+        """Initialize a WeatherFlow sensor entity."""
+        self.device = device
+        self.entity_description = description
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device.serial_number)},
+            manufacturer="WeatherFlow",
+            model=device.model,
+            name=f"{device.model} {device.serial_number}",
```

Comment:
How long is the serial number? It might be nicer to only show the last 4 digits here to avoid long names

Reply:
`{"serial_number":"ST-00084623","type":"rapid_wind","hub_sn":"HB-00068684","ob":[1695413788,1.49,143]}`

its not that long... you think we should still goto 4?

Reply:
Its still a lot to display and 4 is going to be enough to tell apart devices for 99.9% of cases so persona preference would be to shorten it to 4, but if you want to leave it, thats fine as well.

Reply:
We could also drop the device model and just use the serial number since that is the default name the station gets when brought into the Tempest app after all.

Reply:
I dropped the model